### ISSUE TITLE
Add bindings for DataView typed array

### ISSUE THREAD

**Issue by jdm** [servo/rust-mozjs#406]
It will be very similar to the ArrayBufferView typed array bindings.

**Comment by anthgur** [comment]
@jdm I think this might be a prerequisite to https://github.com/servo/servo/issues/20417 so I can take a crack at it too.

**Comment by jdm** [comment]
So it looks like we can't reuse the macro because the JS API for DataView objects functions a little differently. We'll need to hand-write [this implementation](https://github.com/servo/rust-mozjs/blob/master/src/typedarray.rs#L268-L285) to use UnwrapArrayBuffer, [these functions](https://github.com/servo/mozjs/blob/3f38435122538e4b66f88878302f81331c0b6a44/mozjs/js/src/jsfriendapi.h#L2157-L2177), and [this implementation](https://github.com/servo/rust-mozjs/blob/master/src/typedarray.rs#L296-L307) using [this function](https://github.com/servo/mozjs/blob/3f38435122538e4b66f88878302f81331c0b6a44/mozjs/js/src/jsfriendapi.h#L2136-L2144). At that point we should be able to reuse the `array_alias!` macro

**Comment by aditj** [comment]
@jdm  Can I work on this in a attempt to fix servo issue [#20417](https://github.com/servo/servo/issues/20417)

**Comment by aditj** [comment]
> @jdm Can I work on this in an attempt to fix servo/servo issue [#20417](https://github.com/servo/servo/issues/20417)

Also if yes, can you maybe tell me if there a dxr tree for the same? :grin:  :sweat_smile:

**Comment by jdm** [comment]
Absolutely. I'm pretty sure the answer to your DXR question is no, though, since the version of mozjs that we're using is at least 6 months older than the version in DXR/searchfox.

**Comment by aditj** [comment]
@jdm Should I start by creating a macro rule for creating DataView element (like the one typed array element have)?

**Comment by jdm** [comment]
Sure.

**Comment by aditj** [comment]
@jdm 
I think the DataView element would be similar to the typed array element ,any properties that you think should be added?
Also, should I also write DataView Array structure? (And create a seperate array_alias macro for DataView Array)

**Comment by jdm** [comment]
@aditj I do not believe we need a separate DataViewArray structure. We should be able to use the existing TypedArray structure without any changes; we just need to define an implementation of TypedArrayElement and TypedArrayElementCreator that uses the DataView APIs.
